ui/core/components/inputs/consumables.ts

```diff
@@ -0,0 +1,288 @@
+		{ id: ActionId.fromItemId(12662), value: Conjured.ConjuredDemonicRune, showWhen: (p) => p.getLevel() >= 40 },
+	] as Array<IconEnumValueConfig<Player<any>, Conjured>>
+});
+
+export const makeAgilityConsumeInput = makeConsumeInputFactory({consumesFieldName: 'agilityElixir'})
+export const makeFlasksInput = makeConsumeInputFactory({consumesFieldName: 'flask'});
+export const makeFoodInput = makeConsumeInputFactory({consumesFieldName: 'food'});
+export const makeOffHandImbuesInput = makeConsumeInputFactory({consumesFieldName: 'offHandImbue'});
+export const makeMainHandImbuesInput = makeConsumeInputFactory({consumesFieldName: 'mainHandImbue'});
+export const makeStrengthConsumeInput = makeConsumeInputFactory({consumesFieldName: 'strengthBuff'})
+
+export const StrengthBuffInput = makeConsumeInput('strengthBuff', [
+	{ id: ActionId.fromItemId(12451), value: StrengthBuff.JujuPower, minLevel: 46 },
+	{ id: ActionId.fromItemId(9206), value: StrengthBuff.ElixirOfGiants, minLevel: 46 },
+  { id: ActionId.fromItemId(3391), value: StrengthBuff.ElixirOfOgresStrength, minLevel: 20},
+	{ id: ActionId.fromItemId(10310), value: StrengthBuff.ScrollOfStrength },
+] as Array<IconEnumValueConfig<Player<any>, StrengthBuff>>, (p) => p.getLevel() >= 20);
+
+export const SpellDamageBuff = makeConsumeInput('spellPowerBuff', [
+	{ id: ActionId.fromItemId(9155), value: SpellPowerBuff.ArcaneElixir, minLevel: 37 },
+	{ id: ActionId.fromItemId(13454), value: SpellPowerBuff.GreaterArcaneElixir, minLevel: 46 },
+] as Array<IconEnumValueConfig<Player<any>, SpellPowerBuff>>, (p) => p.getLevel() >= 37);
+
+export const FireDamageBuff = makeConsumeInput('firePowerBuff', [
+	{ id: ActionId.fromItemId(6373), value: FirePowerBuff.ElixirOfFirepower, minLevel: 18 },
+	{ id: ActionId.fromItemId(21546), value: FirePowerBuff.ElixirOfGreaterFirepower, minLevel: 40 },
+] as Array<IconEnumValueConfig<Player<any>, FirePowerBuff>>, (p) => p.getLevel() >= 18);
+
+export const ShadowDamageBuff = makeBooleanConsumeInput({id: ActionId.fromItemId(9264), fieldName: 'shadowPowerBuff', minLevel: 40});
+export const FrostDamageBuff = makeBooleanConsumeInput({id: ActionId.fromItemId(17708), fieldName: 'frostPowerBuff', minLevel: 40});
+
+export const FillerExplosiveInput = makeConsumeInput('fillerExplosive', [
+	{ id: ActionId.fromItemId(18641), value: Explosive.ExplosiveDenseDynamite, showWhen: (p) => p.getLevel() >= 40 },
+	{ id: ActionId.fromItemId(15993), value: Explosive.ExplosiveThoriumGrenade, showWhen: (p) => p.getLevel() >= 40 },
+] as Array<IconEnumValueConfig<Player<any>, Explosive>>);
+
+// TODO: These should be moved to consumes
+export const GiftOfArthas = withLabel(
+	makeBooleanDebuffInput({id: ActionId.fromSpellId(11374), fieldName: 'giftOfArthas'}),
+	'Gift of Arthas',
+);
+export const CrystalYield = withLabel(
+	makeBooleanDebuffInput({id: ActionId.fromSpellId(15235), fieldName: 'crystalYield'}),
+	'Crystal Yield',
+);
```

Comment:
I don't think these two should go in consumes, since usually you're modeling someone else in the raid applying these debuffs rather than yourself. For example, DPS sims will often want to check the impact of GoA, but it's the tank who's actually using that consumable.